### PR TITLE
Thread-DB integration for User registration and Metadata caching

### PR DESCRIPTION
**Pull request by Shubham-tiwari123**
- adds user registration to threadDb
- adds procedure contract address and groupId storage on threadDb  
- adds agreement contract address and groupId storage on threadDb  
- adds metaData storage on threadDb

### CODE REVIEW

File: src/database/e2e-encrypt.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/* eslint-disable */
```
Comment (koshikraj) [inline src/database/e2e-encrypt.js]: Move this to utils.

File: src/database/e2e-encrypt.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+export const calculateHash = function(file){
+    return crypto.createHash('sha256').update(file.toString()).digest()
+}
+
+export const convertPass = function(pass){
+    return crypto.createHash('sha256').update(pass.toString()).digest()
+}
+
+export const decryptFile = async function(encryptedData,cipherKey){
```
Comment (koshikraj) [inline src/database/e2e-encrypt.js]: Rename the method to `decryptData`

File: src/database/e2e-encrypt.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        resolve(encryptedData)
+    })
+}
+
+export const calculateHash = function(file){
+    return crypto.createHash('sha256').update(file.toString()).digest()
+}
+
+export const convertPass = function(pass){
```
Comment (koshikraj) [inline src/database/e2e-encrypt.js]: Rename this to a more generic method name.

File: src/database/e2e-encrypt.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            resolve(cipherKey)
+        })
+    }catch (err) {
+        console.error("Error while generating symmetric key:",err)
+        return null
+    }
+}
+
+
+export const encryptFile = function(file,cipherKey){
```
Comment (koshikraj) [inline src/database/e2e-encrypt.js]: Rename the method to `encryptData`

File: src/database/threadDB-utils.js [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+  }catch (err) {
+    throw err
+  }
+}
+
+export const getAllUsers = async function(dbClient,loggedUser){
+  let userKey = e2e.getPublicKey(loggedUser)
+  const threadId = ThreadID.fromBytes(threadDbId)
+  const registeredUsers = await dbClient.find(threadId, 'RegisterUser', {})
+  console.log("UserDetails:",registeredUsers)
```
Comment (koshikraj) [inline src/database/threadDB-utils.js]: Try to clear unnecessary console logs.

**Comment by koshikraj** [comment]
Can you try to merge this branch with `master` and resolve the conflict?

**Comment by Shubham-tiwari123** [comment]
Yes sure, let me make the changes once and then I will try to merge with master

**Pull request merged**